server/bot_api.go

```diff
@@ -245,6 +248,73 @@ func (p *Plugin) handleBotPostRecordings(w http.ResponseWriter, r *http.Request,
 	res.Msg = "success"
 }
 
+func (p *Plugin) handleBotPostJobsStatus(w http.ResponseWriter, r *http.Request, callID, jobID string) {
+	var res httpResponse
+	defer p.httpAudit("handleBotPostJobsStatus", &res, w, r)
+
+	var status public.JobStatus
+	if err := json.NewDecoder(http.MaxBytesReader(w, r.Body, requestBodyMaxSizeBytes)).Decode(&status); err != nil {
+		res.Err = "failed to decode request body: " + err.Error()
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	state, err := p.lockCall(callID)
+	if err != nil {
+		res.Err = fmt.Errorf("failed to lock call: %w", err).Error()
+		res.Code = http.StatusInternalServerError
+		return
+	}
+	defer p.unlockCall(callID)
+
+	if state == nil || state.Call == nil {
+		res.Err = "no call ongoing"
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	if status.JobType == public.JobTypeRecording {
+		if state.Call.Recording == nil {
+			res.Err = "no recording ongoing"
+			res.Code = http.StatusBadRequest
+			return
+		}
+
+		if state.Call.Recording.ID != jobID {
+			res.Err = "invalid recording job ID"
+			res.Code = http.StatusBadRequest
+			return
+		}
+
+		if state.Call.Recording.EndAt > 0 {
+			res.Err = "recording has ended"
+			res.Code = http.StatusBadRequest
+			return
+		}
+
+		state.Call.Recording.EndAt = time.Now().UnixMilli()
+		state.Call.Recording.Err = status.Error
```

Comment:
Maybe I'm misreading, but every api hit will assume you're stopping the recording? We're not using the `Status` field?

Reply:
You are correct. We are only handling one status at the moment so it doesn't matter but yes I definitely need to add a check as we'll be soon using it for the started event as well. I'll send an update :+1: